file/split.rs

```diff
@@ -151,11 +151,8 @@ impl OutputState {
 
     fn write(&mut self, buf: &[u8]) -> io::Result<()> {
         match &mut self.outf {
-            None => {
-                assert!(false);
-                Ok(())
-            }
             Some(ref mut f) => f.write_all(buf),
+            None => Ok(()),
```

Comment:
What is the reason for the assert!()?   Usually these are intentional and should not be removed.

Reply:
The assertion assert!(false) will never be executed.

Reply:
My gut instinct is there is an edge case we are missing.  Nonetheless, going to merge this PR and hope edge case, if any, is caught later.